### PR TITLE
Add aioredis instrumentation

### LINKED ISSUE #1080: Elastic support for aioredis
**Is your feature request related to a problem? Please describe.**
I am using aioredis in my asyncio based application. aioredis is supported by a similar observability platform like newrelic. Now that I have integrated my app with the aioredis library I don't see the logs instrumentation for it.

**Describe the solution you'd like**
Similar to aiopg and asyncpg, there should be instrumentation for aioredis as well

**Describe alternatives you've considered**
N/A

**Additional context**
async libraries are getting much popularity and these should be added for a popular apm solution like elastic.

### PR DESCRIPTION
**Pull request by Sparkycz**
## What does this pull request do?

Adds aioredis instrumentation

## Related issues
closes #1080

### CODE REVIEW

**Comment by Sparkycz** [comment]
I checked the failed jobs and I think these are beyond my changes. Please, anyone, check it out and help me to solve it.

File: tests/instrumentation/asyncio_tests/aioredis_tests.py
Comment (beniwohli) [inline tests/instrumentation/asyncio_tests/aioredis_tests.py:52]: The `black` code formatter fails on this

```suggestion
    conn = await aioredis.create_redis_pool(f"redis://{_host}:{_port}")
```

**Review by beniwohli — COMMENTED** [review]
@Sparkycz awesome stuff, thanks a lot!

The original test failure was a problem on one of our Windows test machines, it went away when I restarted the test run.

The weird @houndci-bot failures seem to be related to https://github.com/houndci/hound/issues/1839. Nothing we can do about that right now...

I'll do a more in-depth review later this week.

**Comment by ajay1mg** [comment]
Thank you @Sparkycz, I checked your code and now wondering how fast you were in adding the support for this. Could you help me understand how you came up with this piece of code?

Going forward I would love to contribute to this since I may have other requirements with my APM integration

File: elasticapm/instrumentation/packages/asyncio/aioredis.py
Comment (beniwohli) [inline elasticapm/instrumentation/packages/asyncio/aioredis.py:48]: Can you use `async_capture_span` here instead? At the moment, it doesn't have a huge impact (all it really does is set the `sync` flag to `False` in the captured meta data), but in the future it will help us capturing additional asyncio data

File: elasticapm/instrumentation/packages/asyncio/aioredis.py
Comment (beniwohli) [inline elasticapm/instrumentation/packages/asyncio/aioredis.py:62]: Also `async_capture_span` :)

**Review by beniwohli — CHANGES_REQUESTED** [review]
@Sparkycz  Great work! I took the liberty to push some needed boilerplate code to ensure your tests are run in our CI test matrix.

I noticed that the maintainers of aioredis are getting ready to release version 2.0.0 of the library, which from the sound of it, changes more or less completely how the API of the library works. We'll probably need to adapt some things. Would you like to do that or should I take over that part?

**Comment by Sparkycz** [comment]
> @Sparkycz Great work! I took the liberty to push some needed boilerplate code to ensure your tests are run in our CI test matrix.
> 
> I noticed that the maintainers of aioredis are getting ready to release version 2.0.0 of the library, which from the sound of it, changes more or less completely how the API of the library works. We'll probably need to adapt some things. Would you like to do that or should I take over that part?

Thanks for the boilerplate code I didn't know It's needed as well..

I'd wait for the stable release and I'll try doing that.. You imagine having two versions of the instrumentation https://github.com/elastic/apm-agent-python/pull/1082/commits/754731f7ab6832f42a5124e5fba1dd1d3b7ab226#diff-523e55d3048a9bdd660f01803427652e28fb5a0610ad41f9c6a98575e83ef67fR22 , don't you?

**Comment by Sparkycz** [comment]
Hello @beniwohli, please, what is the PR waiting for? :-)

**Comment by Sparkycz** [comment]
@basepi Could you unfreeze the PR please?

**Comment by beniwohli** [comment]
@Sparkycz sorry about the wait here, we do have a lot of other work on our plate, so I didn't have a chance to merge the PR yet. I'll get it merged today and do a release soon.

**Pull request merged**
